src/hook_script.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+# rusty-hook
+# version {{VERSION}}
+
+hookName=$(basename "$0")
+gitParams="$*"
+
+if ! command -v rusty-hook >/dev/null 2>&1; then
+  if [ -z "${RUSTY_HOOK_SKIP_AUTO_INSTALL}" ]; then
+    echo "Finalizing rusty-hook configuration..."
+    echo "This may take a few seconds..."
+    cargo install rusty-hook >/dev/null 2>&1
+  else
+    echo "rusty-hook is not installed, and auto install is disabled"
+    echo "skipping $hookName hook"
+    echo "You can reinstall it using 'cargo install rusty-hook' or delete this hook"
+    exit 0
+  fi
+fi
+
+rusty-hook run --hook "$hookName" "$gitParams"
+rhExitCode=$?
+
+if [ $rhExitCode -ne 0 ]; then
+# shellcheck disable=SC2170,SC1083
+  if [ $rhExitCode -eq {{NO_CONFIG_FILE_EXIT_CODE}} ]; then
```

Comment:
`{{NO_CONFIG_FILE_EXIT_CODE}}` is tokenized here so that rusty-hook can inject the correct value as part of writing all the git hook script files. Unsurprisingly, `shellcheck` doesn't like that, hence the above inline disables.